### PR TITLE
Remove AlertManager link in opsgenie and slack templating

### PR DESCRIPTION
**Pull request by marieroque**
## Checklist

towards https://github.com/giantswarm/roadmap/issues/3510
And fix issue with non existing __prometheusurl  variable in alertmanager config

I have:

- [x] Described why this change is being introduced
- [ ] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: files/templates/alertmanager/notification-template.tmpl
Comment (QuentinBisson) [inline files/templates/alertmanager/notification-template.tmpl:16]: Does that work?

Comment (QuentinBisson) [reply files/templates/alertmanager/notification-template.tmpl:16]: I was afraid of the formatting

File: files/templates/alertmanager/notification-template.tmpl
Comment (QuentinBisson) [inline files/templates/alertmanager/notification-template.tmpl:3]: I think we can not define the alertmanager and alertmanager url values if mimir is enabled right?

Comment (marieroque) [reply files/templates/alertmanager/notification-template.tmpl:3]: The alertmanager url is used as a link in the title.
To avoid more refactoring I kept the variable but changed the value in case of mimir.
But I can refactor  properly if you prefer :-)

File: files/templates/alertmanager/notification-template.tmpl
Comment (QuentinBisson) [reply files/templates/alertmanager/notification-template.tmpl:16]: Well if you generate the tests using go test ./... -update then we will see :D

File: files/templates/alertmanager/notification-template.tmpl
Comment (QuentinBisson) [inline files/templates/alertmanager/notification-template.tmpl:22]: I think you forgot the slack.default.titlelink link

File: files/templates/alertmanager/notification-template.tmpl
Comment (QuentinBisson) [reply files/templates/alertmanager/notification-template.tmpl:3]: Well, we should remove it from the slack title link as well :D

File: files/templates/alertmanager/notification-template.tmpl
Comment (marieroque) [reply files/templates/alertmanager/notification-template.tmpl:22]: Well I set the grafana link in __alertmanagerurl variable in case of mimir.

File: files/templates/alertmanager/notification-template.tmpl
Comment (marieroque) [reply files/templates/alertmanager/notification-template.tmpl:3]: Why ?
We can have a direct link to grafana on the title, no ?

File: files/templates/alertmanager/notification-template.tmpl
Comment (QuentinBisson) [reply files/templates/alertmanager/notification-template.tmpl:22]: You're right but I think renaming the variable to alerturl makes more sense than alertmanagerurl?

File: files/templates/alertmanager/notification-template.tmpl
Comment (QuentinBisson) [reply files/templates/alertmanager/notification-template.tmpl:3]: Why would we link to alertmanager as we removed the ingress?

Comment (QuentinBisson) [reply files/templates/alertmanager/notification-template.tmpl:3]: Oh yes that's because you replaced the alertmanagerurl value sure

File: files/templates/alertmanager/notification-template.tmpl
Comment (marieroque) [reply files/templates/alertmanager/notification-template.tmpl:16]: Please don't merge master in my PR, because it creates conflicts with my changes as I didn't know you merged on your side...

File: service/controller/resource/alerting/alertmanagerconfig/test/notification-template/mimir-enabled/capi/case-4-eks.golden
Comment (QuentinBisson) [inline service/controller/resource/alerting/alertmanagerconfig/test/notification-template/mimir-enabled/capi/case-4-eks.golden:15]: I think we probably should change the username here but I'm not sure what happens if we do

File: files/templates/alertmanager/notification-template.tmpl
Comment (QuentinBisson) [inline files/templates/alertmanager/notification-template.tmpl:60]:
```suggestion
👀 Query: {{ template "__queryurl" . }}
```

**Review by QuentinBisson — APPROVED** [review]
LGTM if tested

File: service/controller/resource/alerting/alertmanagerconfig/test/notification-template/mimir-enabled/capi/case-4-eks.golden
Comment (marieroque) [reply service/controller/resource/alerting/alertmanagerconfig/test/notification-template/mimir-enabled/capi/case-4-eks.golden:15]: Me too... 
I think we can keep it because it's still AlertManager that handles alerts.

Comment (QuentinBisson) [reply service/controller/resource/alerting/alertmanagerconfig/test/notification-template/mimir-enabled/capi/case-4-eks.golden:15]: Sure I think with the new postMessage api, this can be what we want anyway and it's not visible anyway :D

File: files/templates/alertmanager/notification-template.tmpl
Comment (marieroque) [reply files/templates/alertmanager/notification-template.tmpl:60]: I proposed `Details` in case of Mimir because the Grafana link is provided information on the query but also on the alert.

Comment (QuentinBisson) [reply files/templates/alertmanager/notification-template.tmpl:60]: Maybe we should rename that Explore then?

**Comment by marieroque** [comment]
✅ Tested on grizzly

Good to know: 
- the title link in #opsgenie slack channel is routing to opsgenie UI
- the title link in #alert-XXX slack channels is routing to AlertManager UI when Mimir is not enabled and to Grafana when Mimir is enabled

File: files/templates/alertmanager/notification-template.tmpl
Comment (marieroque) [reply files/templates/alertmanager/notification-template.tmpl:60]: You prefer query ?

**Pull request merged**
